Urban-Fun/Cells/TimelineCell.m

```diff
@@ -6,21 +6,48 @@
 //
 
 #import "TimelineCell.h"
+#import <MapKit/MapKit.h>
 
 @implementation TimelineCell
 
 - (void)awakeFromNib {
     [super awakeFromNib];
     // Initialization code
+    [self.contentView.layer setBorderColor:(__bridge CGColorRef _Nullable)([UIColor colorNamed:@"Black"])];
 }
 
 - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     [super setSelected:selected animated:animated];
 
     // Configure the view for the selected state
 }
+
+- getCLLocation:(PFGeoPoint *)location{
+    return [[CLLocation alloc] initWithLatitude:location.latitude longitude:location.longitude];
+}
+
 - (void) setTimelineCell{
     self.activityTitleLabel.text = self.activity.title;
     self.activityDescriptionLabel.text = self.activity.activityDescription;
+    CLLocation *activityLocation = [self getCLLocation:self.activity.location];
+    CLLocationDistance distanceFromUser = [self.currentUserLocation distanceFromLocation:activityLocation];
+    CLLocationDistance distanceInMiles = distanceFromUser * 0.000621371;
+    NSNumber *distanceDouble = [NSNumber numberWithDouble:distanceInMiles];
+    if ([distanceDouble intValue] < 10){
+        NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
```

Comment:
Two thing here:
- We can potentially put this into a HelperClass as well since we want to format the distance which is not a responsibility for the cell itself and then we can reuse this code later on in other parts of the code
- Why are you creating two different formatters based on the validation? My 2cents here is that we should use the same formatter for both of them or if you want to validate the distance, then do it inline for the attribute you're changing `formatter.maximumFractionDigits = [distanceDouble intValue] < 10 ? 1 : 0;`

Reply:
Ok, I understand what you are saying for the first bullet point.

For the second bullet point, I moved the "NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];" line that was in both conditional statements and put it on the line above the "if" statement. Is that what you were saying I should do?